package/samplers/auto_sampler/_sampler.py

```diff
@@ -44,14 +45,17 @@
 _SAMPLER_KEY = "auto:sampler"
 # NOTE(nabenabe): The prefix `optuna.` enables us to use the optuna logger externally.
 _logger = get_logger(f"optuna.{__name__}")
+NSGAII_GENERATION_KEY = NSGAIISampler._get_generation_key()
+NSGAIII_GENERATION_KEY = NSGAIIISampler._get_generation_key()
```

Comment:
Just a note: unlike `BaseGASampler.population_size` or the `select_parent` method, `_get_generation_key()` is a private method.I'm not sure if this method will continue to be provided in the future.

The most reliable approach would probably be to make this method public and stabilize the interface. We can discuss this problem as a follow-up.
https://github.com/optuna/optuna/blob/master/optuna/samplers/_ga/_base.py#L42-L43